generators/csharp/codegen/src/context/ProtobufResolver.ts

```diff
@@ -0,0 +1,143 @@
+import {
+    IntermediateRepresentation,
+    ProtobufFile,
+    ProtobufType,
+    TypeId,
+    WellKnownProtobufType
+} from "@fern-fern/ir-sdk/api";
+import { csharp } from "..";
+import { ResolvedWellKnownProtobufType } from "../ResolvedWellKnownProtobufType";
+import { CsharpTypeMapper } from "./CsharpTypeMapper";
+
+export class ProtobufResolver {
+    private ir: IntermediateRepresentation;
+    private csharpTypeMapper: CsharpTypeMapper;
+
+    public constructor(ir: IntermediateRepresentation, csharpTypeMapper: CsharpTypeMapper) {
+        this.ir = ir;
+        this.csharpTypeMapper = csharpTypeMapper;
+    }
+
+    public getNamespaceFromProtobufFileOrThrow(protobufFile: ProtobufFile): string {
+        const namespace = protobufFile.options?.csharp?.namespace;
+        if (namespace == null) {
+            throw new Error(
+                `The 'csharp_namespace' file option must be declared in Protobuf file ${protobufFile.filepath}`
+            );
+        }
+        return namespace;
+    }
+
+    public getProtobufStructTypeOrThrow(): csharp.Type {
+        const resolvedType = this.getProtobufStructType();
+        if (resolvedType === undefined) {
```

Comment:
why `=== undefined` here instead of `== null`?

Reply:
Good catch, we should definitely use `== null` 👍